Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -522,8 +522,32 @@ extension RouteController: Router {
         
         // If we still wait for the first status from NavNative, there is no need to reroute
         guard let status = status ?? navigator.getStatus() else { return true }
-        let offRoute = status.routeState == .offRoute || status.routeState == .invalid
-        return !offRoute
+
+        /// NavNative doesn't support reroutes after arrival.
+        /// The code below is a port of logic from LegacyRouteController
+        /// This should be removed once NavNative adds support for reroutes after arrival. 
+        if status.routeState == .complete {
```

Comment:
If route state is changed to completed, NavNative won't detect off routes anymore, so we drop to temporary and straightforward logic. If the user moves far enough from the destination point, trigger the reroute if the delegate allows that.